app/code/core/Mage/Core/Model/Translate/Config.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Maho
+ *
+ * @category   Mage
+ * @package    Mage_Core
+ * @copyright  Copyright (c) 2024 Maho (https://mahocommerce.com)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Config model for loading jstranslate.xml files
+ *
+ * @category   Mage
+ * @package    Mage_Core
+ */
+class Mage_Core_Model_Translate_Config extends Varien_Simplexml_Config
+{
+    /**
+     * Extends current node with xml from $source, appending nodes without merging
+     *
+     * @param Varien_Simplexml_Config $config
+     * @param boolean $overwrite Argument has no effect, included for PHP 7.2 method signature compatibility
+     * @return $this
+     */
+    public function extend(Varien_Simplexml_Config $config, $overwrite = true)
```

Comment:
```suggestion
     * @param Varien_Simplexml_Config $config
     * @param boolean $overwrite Argument has no effect, included for PHP 7.2 method signature compatibility
     * @return $this
     */
    public function extend(Varien_Simplexml_Config $config, $overwrite = true)
```

I guess we can't add typing here? but also, we don't support php 7 so I don't know if that changes anything

Reply:
If I change the method signature to:

```php
public function extend(Varien_Simplexml_Config $config, bool $overwrite = true)
```

I get this PHPStan error:

```
 ------ ------------------------------------------------------------------------------------------------------------------------- 
  Line   app/code/core/Mage/Core/Model/Translate/Config.php
 ------ ------------------------------------------------------------------------------------------------------------------------- 
  22     Parameter #2 $overwrite (bool) of method Mage_Core_Model_Translate_Config::extend() is not contravariant with parameter  
         #2 $overwrite (mixed) of method Varien_Simplexml_Config::extend().
 ------ ------------------------------------------------------------------------------------------------------------------------- 
```

---

Regarding the comment, I guess it should really say "included for PHP 7.2 and above method signature", but I just used the same comment as in some other classes: 

https://github.com/search?q=repo%3AMahoCommerce%2Fmaho%20Argument%20has%20no%20effect%2C%20included%20for%20PHP%207.2%20method%20signature%20compatibility&type=code

Reply:
maybe we could do an @inheritDoc instead so we don't copy the docblock?

Reply:
Related question. These both would work, right?

```php
////////////////////
// Option 1:

/** @inheritDoc */
#[\Override]
public function someFunction() {}

////////////////////
// Option 2:

/** 
 * @inheritDoc
 */
#[\Override]
public function someFunction() {}
```

I see the second one in the source  code, but it's so long.... 😅

Reply:
Also.... I don't fully understand the point of `@inheritDoc` in the first place. PHPStan does it automatically if there's no docblock at all, but maybe it's some IDEs that need it?

Reply:
yeah it's unnecessary long, and there are a ton of useless docblock anyway like `/** constructor`. I also seem to have noticed that probably @inheritDoc is not even necessary anymore (I noticed it in phpstorm but I didn't really test for real)

Reply:
> Also.... I don't fully understand the point of `@inheritDoc` in the first place. PHPStan does it automatically if there's no docblock at all, but maybe it's some IDEs that need it?

probably it's just an old thing not necessary anymore, i've asked AI and it said:

> No, @inheritdoc and {@inheritDoc} tags are largely considered unnecessary in PHP documentation as of 2024. Modern PHP tooling has evolved to handle inheritance automatically without requiring explicit inheritance tags7.
> Symfony has abandoned the usage of @inheritDoc as of version 6.25
> Modern PHP documentation tools can determine inheritance independently7
> The distinction between @inheritDoc and {@inheritDoc} has been a source of confusion even for experienced developers3

Reply:
so, in phpstorm @inheritdoc seems to be useless.

I've removed a bunch of them and phpstan/phpcs didn't care at all, should we trash them all?

Reply:
I'd say inheritdoc is officially useless, i've used this code

```php
<?php

class test1
{
    /**
     * This method is nice!
     */
    public function method1(string $a, int $b): string
    {
        return $a . (string) $b;
    }
}

class test2 extends test1
{
    public function method1(string $a, int $b): string
    {
        return "inherited" . $a . (string) $b;
    }
}

$test = new test1();
$test->method1("a", 1);

$test2 = new test2();
$test2->method1();
```

and hovering the overridden method, the description of the base one is visible:
<img width="515" alt="Screenshot 2024-11-22 alle 13 46 32" src="https://github.com/user-attachments/assets/8e803599-bf6d-4b4d-87b2-f6e436b9d3f4">

Reply:
removed all in commit 6c05282a79